TilerServer/TilerWebModule.pas

```diff
@@ -3248,6 +3274,236 @@ function TSliceLocation.HandleSliceUpdate(const aBuffer: TByteBuffer; var aCurso
+      setLength(polygon, 4);
+      for isgop in fLocations do
+      begin
+        point := GeometryToPoint(aExtent, aPixelWidth, aPixelHeight, isgop.Value.lcoation);
+        rect.Create(point);
+        rect.Inflate(pieRadius, pieRadius); //Hardcoded radius values for the junctions
+        if (rect.Right>=0) and (rect.Bottom>=0) and (rect.Left<=aBitmap.Width) and (rect.Top<=aBitmap.Height) then
+        begin
+          colors := fPalette.ValueToColors(isgop.Value.value);
+          if colors.fillColor<>0 then
+          begin
+            radiusPoint := PointF(pieRadius, pieRadius);
+            aBitmap.Canvas.Fill.Color := TAlphaColorRec.White;
+            aBitmap.Canvas.FillArc(point, radiusPoint, 0, 360, 1);
+
+            startAngle := -90;
+            sweepAngle := isgop.Value.value * 360;
+
+            aBitmap.Canvas.Fill.Color := colors.fillColor;
+            aBitmap.Canvas.FillArc(point, radiusPoint, startAngle, sweepAngle, 1);
+
+            startAngle := 90;
+            stopAngle := (360 - sweepAngle) + startAngle;
+
+            polygon[0].X := point.X + Round((pieRadius + 1) * Cos(DegToRad(startAngle)));
+            polygon[0].Y := point.Y - Round((pieRadius + 1) * Sin(DegToRad(startAngle)));
+            polygon[1].X := point.X;
+            polygon[1].Y := point.Y;
+            polygon[2].X := point.X + Round((pieRadius + 1) * Cos(DegToRad(stopAngle)));
+            polygon[2].Y := point.Y - Round((pieRadius + 1) * Sin(DegToRad(stopAngle)));
+            polygon[3].X := polygon[0].X;
+            polygon[3].Y := polygon[0].Y;
+
+            if sweepAngle <= 180 then
+              polyColor := colors.fillColor
+            else
+              polyColor := TAlphaColorRec.White;
+
+            if sweepAngle <360 then
+            begin
+              aBitmap.Canvas.Fill.Color := polyColor;
+              aBitmap.Canvas.FillPolygon(polygon, 1);
+            end;
+          end;
+
+          if colors.outlineColor<>0 then
+          begin
+            aBitmap.Canvas.Stroke.Color := colors.outlineColor;
+            aBitmap.Canvas.Stroke.Thickness := 1; // todo: default width?
+            aBitmap.Canvas.DrawEllipse(rect, 1);
+          end
+          else
+          begin
+            aBitmap.Canvas.Stroke.Color := TAlphaColorRec.Lightgray;
+            aBitmap.Canvas.Stroke.Thickness := 1;
+            aBitmap.Canvas.DrawEllipse(rect, 1);
+          end;
+        end;
+      end;
+      Result := gtsOk;
+    finally
+      aBitmap.Canvas.EndScene;
+    end;
+  end
+  else Log.WriteLn('TSliceJunctionsPie layer '+fLayer.LayerID.ToString+': no palette defined', llError);
+end;
+
+function TSliceJunctionsPie.getDataValueAtPoint(const aLat, aLon: Double; var aValue: Double): TGenerateTileStatus;
+var
+  isgop: TPair<TWDID, TSliceLocationObject>;
+begin
+  fDataLock.BeginRead;
+  try
+    for isgop in fLocations do
+    begin
+      if isgop.Value.extent.Contains(aLon, aLat) then

```

Comment:
Does extent have width and height or does uit only contain the point. If the latter is true then Contains will never return true and maybe an inline inflate with a small delta would make it work?

Reply:
fixed